### PR TITLE
Revisited the way map layers authentication is handled in the UI.

### PR DESCRIPTION
**Pull request by mdastous-bentley**
No longer display username/password fields by default  in the custom map layers dialog:  If validation fails and reports authentication is needed, we then ask end-user for credentials.

### CODE REVIEW

File: common/changes/@itwin/core-frontend/geo-maplayers_auth_refactor_2022-01-10-14-48.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/core-frontend",
+      "comment": "Revisited the way map layers authentication is handled in the UI.",
```
Comment (pmconne) [inline common/changes/@itwin/core-frontend/geo-maplayers_auth_refactor_2022-01-10-14-48.json:5]: Can you reword this in a way that means something to consumers of the package? i.e., be more specific about what actually changed?

Comment (mdastous-bentley) [reply common/changes/@itwin/core-frontend/geo-maplayers_auth_refactor_2022-01-10-14-48.json]: You are right, updated it to use the PR description instead.  Does that make sense to you?

File: core/frontend/src/tile/map/MapLayerAuthentication.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module DisplayStyles
```
Comment (pmconne) [inline core/frontend/src/tile/map/MapLayerAuthentication.ts:6]: Use the name of a module defined in core-frontend.ts.

**Review by MarcBedard8 — APPROVED** [review]
Look good

**Review by pmconne — CHANGES_REQUESTED** [review]
If you're going to add a dedicated MapLayers doc group please update all files that should belong to that group. Currently it's a bit of a mess: MapLayerImageryFormats.ts specifies `@module Tiles`, MapLayerFormatRegistry.ts specifies `@module Views`, etc.

**Comment by mdastous-bentley** [comment]
> If you're going to add a dedicated MapLayers doc group please update all files that should belong to that group. Currently it's a bit of a mess: MapLayerImageryFormats.ts specifies `@module Tiles`, MapLayerFormatRegistry.ts specifies `@module Views`, etc.

Agreed, I updated files that I felt that was purely related to maplayers.  I feel like the remaining Map* classes are more specific to tiling stuff.

**Review by pmconne — APPROVED** [review]

**Comment by mdastous-bentley** [comment]
@calebmshafer Can you please promote @MarcBedard8  as code owner for maplayers?

**Comment by calebmshafer** [comment]
> @calebmshafer Can you please promote @MarcBedard8 as code owner for maplayers?

@mdastous-bentley feel free to open up a PR with the change and we can approve.

**Review by bsteinbk — APPROVED** [review]

**Pull request merged**